test/integration/dispatcher_integration_test.cc

```diff
@@ -177,13 +177,107 @@ TEST_P(DispatcherIntegrationTest, Basic) {
   ASSERT_EQ(cc.on_headers_calls, 1);
   ASSERT_EQ(cc.on_data_calls, 2);
   ASSERT_EQ(cc.on_complete_calls, 1);
+
+  test_server_->waitForCounterEq("http.dispatcher.stream_success", 1);
+}
+
+TEST_P(DispatcherIntegrationTest, BasicNon2xx) {
```

Comment:
Maybe we should have a positive case also: `Basic2xx`?

Reply:
Do we want a test case to ensure we are incrementing the stats at the appropriate time based on: https://github.com/lyft/envoy-mobile/pull/871/files#r433531576

Reply:
> Maybe we should have a positive case also: `Basic2xx`?

`Basic` covers that.

Reply:
> Do we want a test case to ensure we are incrementing the stats at the appropriate time based on

I believe the race test below does that. Let me double check

Reply:
https://github.com/lyft/envoy-mobile/pull/871/files/bee32cbaa7147d723adc4ccabd810f65192efa65#diff-bb212e49f4cf7a1c832af632465e1440R339 this test shows that even if we received headers if cancelation won out the race we get the right stat